docs/custom_integrations.md

```diff
@@ -0,0 +1,18 @@
+# Custom integrations
+
+BlockHound can be extended without changing its code by using
+[the JVM's SPI mechanism](https://docs.oracle.com/javase/7/docs/api/java/util/ServiceLoader.html).
+
+You will need to implement `reactor.blockhound.integration.BlockHoundIntegration` interface
+and add the implementor to `META-INF/services/reactor.blockhound.integration.BlockHoundIntegration` file.
+
+**Hint:** consider using [Google's AutoFactory](https://github.com/google/auto/tree/master/factory) for it:
+```java
+@AutoFactory(BlockHoundIntegration.class)
+public class MyIntegration implements BlockHoundIntegration {
+    // ...
+}
+```
+
```

Comment:
Maybe add a paragraph or two of examples of what you can do in an integration. Eg. say that you can access the builder and do everything that was discussed in "customize" section, but in addition you can add library-specific code (using hooks and listeners the library offers, or extending classes of the library). Then link to blockhound default integrations as "Blockhound's default integration use the same mecanism, have a look [here] if you want more advanced examples.".

Reply:
Done 👍